File: binary_search.py
import numpy as np


def binary_search(list, target):
    low = 0
    high = len(list)

    while low <= high:
        mid = (low + high) //2
        if list[mid] == target:
            print(f"Target is found in index {mid}")
            return 1
        elif list[mid] > target:
            high = mid
            mid+=1
        elif list[mid] < target:
            low = mid
            mid+=1
        else:
            print("Target is not found")


if __name__ == '__main__':
    lis = list(set(np.random.randint(1000,size=400)))
    #lis.sort()
    target = 766
    print(lis)
    binary_search(lis,target)

File: binary_search.cpp
#include <iostream>
#include <vector>

int binary_search(std::vector<int> &vec, int target){
    int low = 0;
    int high = vec.size()-1;
    //int mid =  (high + low)/2;
    
    while (low <= high){
        int mid = (high + low)/2;
        if (target == vec[mid]){
            std::cout<<"Target is found in "<<mid<<std::endl;
            return 0;
        }
        else if (target < vec[mid]){
            high = mid;
            mid++;
        }
        else if (target > vec[mid]){
            low = mid;
            mid++;
        }
    }
    std::cout<<"target not found"<<std::endl;
    return -1;
}


int main(){
    std::vector<int> vec = {1,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22, 23};
    binary_search(vec,22);
    return 0;
}